### ISSUE TITLE
The Value() call always get value from previous page after switching to a new page with Navigate() or Click()

### ISSUE THREAD

**Issue by ccll** [chromedp/chromedp#152]
I'm scraping contents from multiple pages one by one, each of them have the same DOM structure.
The logic is:
```
Navigate("page-1")
WaitVisible("the-element")
Value("the-element", &value)
...
Navigate("page-2")    // or Click("next-page-button")
WaitVisible("the-element")
Value("the-element", &value)
```

I thought after navigating to 'page-2', the WaitVisible() should block until the element is loaded and visible, but actually it's not, the WaitVisible() call returns immediately, and the Value() call just got the old value from 'page-1'.

I guess it's because the DOM does not reset immediately after Navigate() or Click(), thus all the actions still operates on the previous DOM, right?

Currently it seems only one solution to this problem: Sleep() **long enough** for 'page-2' to be fully loaded.
This is too bad as **long enough** is not a defined behavior.

**Comment by kenshaw** [comment]
There are currently issues with working with "single page applications" and this package. As a temporary workaround, I would put a `Sleep` action in manually after every page navigation. When I designed / wrote the initial version, I did not think this was going to be a problem, but obviously it has been. The fix is not simple, as it requires listening for all the various network events and killing the frame's nodes. At the moment, the current target handler only keeps one frame's nodes in state. The answer to this is to listen to all network events, and to keep multiple target handlers, network loaders, network requests, pages, frames, and nodes all in state simultaneously.

Let's just mention all of those dimensions here:

There is a 1:many relationship for all of these n => m:
```
   Browser => target => network loaders => requests => pages => frames => nodes
```

Let's not even get into the "timeline" features ...
